src/DemuxTopic.cpp

```diff
@@ -6,105 +6,38 @@
 
 namespace FileWriter {
 
-ProcessMessageResult ProcessMessageResult::OK(int64_t ts) {
-  if (ts < 0) {
-    throw std::runtime_error("Invalid timestamp");
-  }
-  ProcessMessageResult ret;
-  ret._ts = ts;
-  return ret;
-}
-
-ProcessMessageResult ProcessMessageResult::ERR() {
-  ProcessMessageResult ret;
-  ret._ts = -1;
-  return ret;
-}
-
-ProcessMessageResult ProcessMessageResult::ALL_SOURCES_FULL() {
-  ProcessMessageResult ret;
-  ret._ts = -2;
-  return ret;
-}
-
-ProcessMessageResult ProcessMessageResult::STOP() {
-  ProcessMessageResult ret;
-  ret._ts = -3;
-  return ret;
-}
-
-DemuxTopic::DemuxTopic(std::string topic)
-    : _topic(topic), _stop_time(std::numeric_limits<uint64_t>::max()) {}
+DemuxTopic::DemuxTopic(std::string TopicName) : Topic(TopicName) {}
 
 DemuxTopic::~DemuxTopic() { LOG(Sev::Debug, "dtor"); }
 
 DemuxTopic::DemuxTopic(DemuxTopic &&x) { swap(*this, x); }
 
 void swap(DemuxTopic &x, DemuxTopic &y) {
   using std::swap;
-  swap(x._topic, y._topic);
-  swap(x._sources_map, y._sources_map);
-  swap(x._stop_time, y._stop_time);
-}
-
-DemuxTopic::DT DemuxTopic::time_difference_from_message(Msg const &msg) {
-  auto &reader = FlatbufferReaderRegistry::find(msg);
-  if (!reader) {
-    LOG(Sev::Error, "ERROR unknown schema id?");
-    return DT::ERR();
-  }
-  auto srcn = reader->source_name(msg);
-  return DT(srcn, reader->timestamp(msg));
+  swap(x.Topic, y.Topic);
```

Comment:
I thought we were going to fully qualify this and lose the using.

Reply:
Sorry, forgot about that one. This is now fixed.